kubespawner/objects.py

```diff
@@ -236,6 +245,41 @@ def make_pod(
         image_secret.name = image_pull_secret
         pod.spec.image_pull_secrets.append(image_secret)
 
+    if ssl_enabled(env):
+        key_name = 'user-'+env['JUPYTERHUB_USER']
```

Comment:
I don't understand the first point.  I'm not inserting anything into the env.  I'm using the data that is being passed in by jupyter.  I also don't understand what this line has to do with the secret name?  I can certainly make it configurable, but what do we gain by that?  This creates the secret so what value do we get by making the secret name configurable?

If the goal is to move this logic from objects.py to spawner.py I can certainly do that, but the env does already contain this information.  Let me know what you think is best.

Reply:
Ah, so 'env' should be treated as a dumb dictionary of stuff to be passed through to the podspec - we shouldn't reach into it to grab stuff. This function should instead get all the info it needs as explicit parameters, rather than implicitly reach into env.

Reply:
We should make the secret object's name configurable to match the fact that [pvc](https://github.com/jupyterhub/kubespawner/blob/master/kubespawner/spawner.py#L334) and [pod names](https://github.com/jupyterhub/kubespawner/blob/master/kubespawner/spawner.py#L309) are also configurable.

Reply:
I will do both of those things as requested.

I don't understand the purpose of not tying ourselves to what comes from Jupyter and instead of making the user duplicate configure the values, but I'll do it.

I also don't understand the value of allowing the secret name to be configurable.

Reply:
> I also don't understand the value of allowing the secret name to be configurable.

Quoting the doc string from https://github.com/jupyterhub/kubespawner/blob/master/kubespawner/spawner.py#L309:

> This must be unique within the namespace the pods are being spawned
> in, so if you are running multiple jupyterhubs spawning in the
> same namespace, consider setting this to be something more unique.

Reply:
Which is why the secret was using the same format as the pod which is always unique.  Hence why I don't know why it needs to be configurable.